pkgs/gnolang/machine.go

```diff
@@ -99,26 +115,27 @@ func NewMachineWithOptions(opts MachineOptions) *Machine {
 		}
 	}
 	context := opts.Context
-	mm := &Machine{
-		Ops:        make([]Op, 1024),
-		NumOps:     0,
-		Values:     make([]TypedValue, 1024),
-		NumValues:  0,
-		Package:    pv,
-		Alloc:      alloc,
-		CheckTypes: checkTypes,
-		ReadOnly:   readOnly,
-		MaxCycles:  maxCycles,
-		Output:     output,
-		Store:      store,
-		Context:    context,
-	}
+	mm := machinePool.Get().(*Machine)
```

Comment:
This seems dangerous because if we add new state to a Machine, it may not get cleared upon Machine re-use. For example, NumOps may not be zero for a machine that panic'd or something similar.

Machine.Clear() could work but programmers can still forget to Clear(). I don't see a better way though.

Reply:
There is already a method that does that but it's called 'Release'

Reply:
No. This way, we would lose most of the benefits because it is not keeping the previous slice allocation.
The automatic default value of a slice is `nil`.  So every time we create a machine, we would have the previous problem of allocating 2 slices of 1024 bytes.

Reply:
can't we modify it to preserve the previous slice allocation?
isn't that what it's doing with Ops: mm.ops?

Reply:
(it is using the same slices referenced from the pool's Machine, but using the Constructor{} pattern to reset unmentioned fields.)

Reply:
You want is to omit `machinePool.Put()`, if i understand correctly.
In that case, what would happen is no instances will be reused via using the pool.
`machinePool.Get()` will always call the constructor and make new allocations.
Once an object is out of the pool, by calling `Get()`, it is no longer available in the pool until you call `Put()`

Reply:
No, I mean like this: https://github.com/gnolang/gno/pull/672/files